### PR TITLE
Add docs for memoization

### PR DESCRIPTION
**Pull request by sfalmo**
Add documentation in FAQ as discussed in #131.

### CODE REVIEW

File: docs/src/faq.md [hunk 1]
```diff
@@ -114,3 +114,13 @@ In case these operators are too complicated to code them yourself, here are a fe
...
 
 An alternative is differentiating through the KKT conditions, which is exactly what [DiffOpt.jl](https://github.com/jump-dev/DiffOpt.jl) does for JuMP models.
+
+### Memoization
+
+In some cases, performance might be increased by using memoization to prevent redundant calls to `forward`.
+For instance, this is relevant when calculating large Jacobians with forward differentiation, where the computation happens in [chunks](https://juliadiff.org/ForwardDiff.jl/stable/user/advanced/#Configuring-Chunk-Size).
+Packages such as [Memoize.jl](https://github.com/JuliaCollections/Memoize.jl) and [Memoization.jl](https://github.com/marius311/Memoization.jl) are useful for defining a memoized version of `forward`:
+
+```julia
```
Comment (gdalle) [inline docs/src/faq.md:124]: which package one are you using here?

**Review by gdalle — COMMENTED** [review]
Thank

File: docs/src/faq.md (hunk 1 above)
Comment (gdalle) [reply docs/src/faq.md:124]: I was thinking we might wanna specify the import in the example

Comment (sfalmo) [reply docs/src/faq.md:124]: Sure, I have updated the PR

**Comment by sfalmo** [comment]
How funny, [v0.15.11](https://github.com/jonniedie/ComponentArrays.jl/releases/tag/v0.15.11) of ComponentArrays.jl, which was released 2h ago, now provides an rrule that makes Zygote work. Hence [this](https://github.com/gdalle/ImplicitDifferentiation.jl/blob/74e1a86d793753605836814c719263f6f7d01f01/examples/3_tricks.jl#L67) @test_broken check actually passes.
I think we should address this in a new PR.

**Comment by gdalle** [comment]
Wow, that's fun indeed ^^ wanna do the honors?

**Comment by gdalle** [comment]
Ok we're good, I also did a bit of cleanup. Wanna merge main into this?

**Comment by gdalle** [comment]
The diff is huge now, can you merge main into this branch to highlight only your changes to the docs?

**Comment by sfalmo** [comment]
Sorry, did some weird rebasing by accident. Now it should be fine.

**Review by gdalle — APPROVED** [review]
all good, thanks!

**Pull request merged**